packages/ui/src/composables/prop-type-filter.ts

```diff
@@ -0,0 +1,22 @@
+import { ComputedRef } from 'vue'
+
+const formatToLog = (value: unknown) => (typeof value === 'string' ? `'${value}'` : JSON.stringify(value))
+
+export function usePropTypeFilter<T extends object>(props: T) {
+  return function <K extends keyof T>(name: K, validValues: readonly T[K][], defaultValue: T[K]): ComputedRef<T[K]> {
+    return computed(() => {
```

Comment:
Is there a real need to create scoped function here? Why not just a single composable that always accepts props, prop name etc? It would be more consistent with e.g. `useVModel()`